pkg/controller/plan/controller.go

```diff
@@ -229,9 +246,28 @@ func (r Reconciler) Reconcile(request reconcile.Request) (result reconcile.Resul
 		return
 	}
 
-	// Archive the plan.
+	// Archive before deleting the plan.
+	if !plan.DeletionTimestamp.IsZero() {
+		r.archive(plan)
+		plan.RemoveFinalizer(ArchiveFinalizer)
+		err = r.Update(context.TODO(), plan)
+		if err != nil {
+			return
+		}
+		return
+	}
+
+	// Archive the plan due to a user request.
 	if plan.Spec.Archived {
 		r.archive(plan)
+		plan.Status.SetCondition(
```

Comment:
Setting the condition now happens outside of `archive()` and only when the archive is user requested rather than prompted by deletion. This is because we can't do anything except remove finalizers once the deletion timestamp is set.